kclvm/tools/src/LSP/src/goto_def.rs

```diff
@@ -0,0 +1,180 @@
+use std::fs;
+use std::path::Path;
+
+use kclvm_ast::ast::{AssignStmt, ImportStmt, Program, Stmt, UnificationStmt};
+use kclvm_ast::pos::ContainsPos;
+use kclvm_config::modfile::KCL_FILE_EXTENSION;
+use kclvm_error::Position as KCLPos;
+use kclvm_sema::resolver::scope::ProgramScope;
+use lsp_types::{GotoDefinitionResponse, Url};
+use lsp_types::{Location, Range};
+
+use crate::to_lsp::lsp_pos;
+
+pub(crate) fn goto_definition(
+    program: Program,
+    kcl_pos: KCLPos,
+    prog_scope: ProgramScope,
+) -> Option<lsp_types::GotoDefinitionResponse> {
+    match program.pos_to_stmt(&kcl_pos) {
+        Some(node) => match node.node {
+            Stmt::Unification(stmt) => goto_def_for_unification(stmt, kcl_pos, prog_scope),
+            Stmt::Assign(stmt) => goto_def_for_assign(stmt, kcl_pos, prog_scope),
+            Stmt::Import(stmt) => goto_def_for_import(stmt, kcl_pos, prog_scope, program),
+            _ => None,
+        },
+        None => None,
+    }
+}
+
+pub(crate) fn find_name_in_program_scope(
+    name: &String,
+    prog_scope: ProgramScope,
+) -> Vec<(KCLPos, KCLPos)> {
+    let mut positions = vec![];
+    let mut scopes = vec![];
+    for s in prog_scope.scope_map.values() {
+        scopes.push(s.borrow().clone());
+    }
+
+    while !scopes.is_empty() {
+        let s = scopes.pop().unwrap();
+        match s.lookup(&name) {
+            Some(obj) => {
+                let obj = obj.borrow().clone();
+                positions.push((obj.start, obj.end));
+            }
+            None => {
+                for c in s.children {
+                    scopes.push(c.borrow().clone());
+                }
+            }
+        }
+    }
+    positions
+}
+
+pub(crate) fn positions_to_goto_def_resp(
+    positions: &Vec<(KCLPos, KCLPos)>,
```

Comment:
s/&Vec<(KCLPos, KCLPos)>/&[(KCLPos, KCLPos)]

Use `cargo clippy` to fix more such issues.